### ISSUE TITLE
create API's for the opportunities with express.js 

### ISSUE THREAD

**Issue by hparashar27** [hparashar27/OpenlySource#17]
opportunities is one of the important part of openlysource , here you get all the information regarding upcoming open source opportunities like gsoc , sob , mlh fellowship and many more open source organization's events 

for making these apis you can take help from this video description
- https://www.youtube.com/watch?v=x4hsV_q_YQc&t=1479s

or for more such programs you can take help from this github repo-
https://github.com/deepanshu1422/List-Of-Open-Source-Internships-Programs

So, the api contain all such fields such as 
* URL of the logo or icon of event 
* Name of event 
* Description of event
* Month in which the event is going to  start
* URL of the Event website

So, Basically  opportunities page contains all such events information..

Anyone can start working on it 
All the best

**Comment by hparashar27** [comment]
So, for now i make dummy frontend for the opportunities page -
it looks like

https://user-images.githubusercontent.com/96375055/215266365-d8dc4776-1e23-4725-9e73-f8277f845748.mp4

**Comment by Faizan-omer** [comment]
Hey I would like to work on this can you assign me to it? Also do we only need a GET api or a POST api as well. Please share more details

**Comment by hparashar27** [comment]
actually we need both type of API's GET  as well as POST 
But for now we need GET API's only so that it can be rendered on the opportunities frontend . later we can think about POST apis

**Comment by Faizan-omer** [comment]
Great. I can get started on this where do I get the data to store and fetch from the express app.  I can set up an express application does it need to be on a separate repository?

**Comment by hparashar27** [comment]
if you can able to manage to do in this same repo then it will great
make a folder named "openlysource-backend" in this repo and setup your server.js and app.js

**Comment by Faizan-omer** [comment]
yes ofcourse, I'll fork the repo please assign this to me, also I might need the data for the opportunities/programs where can I get that? Please also clarify if you want to use mongodb or SQL database for backend.

**Comment by hparashar27** [comment]
use MongoDB for database and you get the data from

for making these apis you can take help from this video description
https://www.youtube.com/watch?v=x4hsV_q_YQc&t=1479s

or for more such programs you can take help from this github repo-
https://github.com/deepanshu1422/List-Of-Open-Source-Internships-Programs

**Comment by Faizan-omer** [comment]
Hi @hparashar27 , I created a PR kindly review it and let me know if there are any issues. Thanks.

**Comment by hparashar27** [comment]
@Faizan-omer can you tell how can we call get and post requests like i am trying to call get apis but it didnt work in this way 
![Postman_lK9WhCX5NB](https://user-images.githubusercontent.com/96375055/216079633-c5e74e9c-f394-4a3a-b76c-3f2c044fd0c6.png)

So can you tell us to which route i should send get and post request to get data

**Comment by Faizan-omer** [comment]
You're using the wrong endpoint route. it is supposed to be `localhost:8000/api/opportunities/events` let me know if it doesn't work.

**Comment by hparashar27** [comment]
@Faizan-omer can you also make deleteEvent and updateEvent Apis as well

**Comment by Faizan-omer** [comment]
Sure assign me the task and I'll look into it

**Comment by hparashar27** [comment]
completed
